File: #0330.patching-array.cpp
class Solution {
public:
    int minPatches(vector<int>& nums, int n) {

        int size = nums.size();
        int i = 0;
        long num = 0;
        int result = 0;
        while (num < n) {
            if (i < size) {
                if (nums[i] <= num + 1) {
                    num += nums[i];
                    i++;
                } else {
                    result++;
                    num += num + 1;
                }
            } else {
                result ++;
                num += num + 1;
            }
        }
        return result;
    }
};

File: #0330.patching-array.js
/**
 * @param {number[]} nums
 * @param {number} n
 * @return {number}
 */
var minPatches = function(nums, n) {
    let result = 0;
    let i = 0;
    let sum = 0;
    while (sum < n) {
        if (i < nums.length && nums[i] <= sum + 1) {
            sum += nums[i];
            i++;
        } else {
            sum += sum + 1;
            result++;
        }
    }
    return result;
};
